crates/pevm/tests/mainnet.rs

```diff
@@ -57,3 +57,44 @@ fn mainnet_blocks_from_disk() {
         }
     });
 }
+
+#[tokio::test(flavor = "multi_thread")]
+#[cfg(all(feature = "rpc-storage", feature = "optimism"))]
+async fn optimism_mainnet_blocks_from_rpc() {
+    use alloy_provider::{Provider, ProviderBuilder};
+    use alloy_rpc_types_eth::{BlockId, Block};
+    use pevm::chain::{PevmChain, PevmOptimism};
+    use op_alloy_rpc_types::Transaction as OpTransaction;
+    use std::borrow::Cow;
+    use serde_json::json;
+
+    let rpc_url = match std::env::var("OPTIMISM_RPC_URL") {
+        Ok(value) if !value.is_empty() => value.parse().unwrap(),
+        _ => reqwest::Url::parse("https://mainnet.optimism.io").unwrap(),
+    };
+
+    // First block under 50 transactions of each EVM-spec-changing fork
+    for block_number in [
+        105235063, // REGOLITH
+        105235064, // First block after REGOLITH
+        105235065, // Second block after REGOLITH
```

Comment:
OP-Regolith doesn't change EVM AFAIK:
https://specs.optimism.io/protocol/regolith/overview.html

Could you better add Canyon, Ecotone, Fjord, Granite blocks?

```suggestion
        114874075, // CANYON (https://specs.optimism.io/protocol/canyon/overview.html)
                  // TODO: doesn't pass `Err(ExecutionError("Database(InvalidNonce(0))"))`
                  // 117874236, // ECOTONE (https://specs.optimism.io/protocol/ecotone/overview.html)
                  // 122874325, // FJORD (https://specs.optimism.io/protocol/fjord/overview.html)
                  // 125874340, // GRANITE (https://specs.optimism.io/protocol/granite/overview.html)
```